### ISSUE TITLE
[Diagnostics] Clang should issue a diagnostic when only one global operator delete is defined

### ISSUE THREAD

**Issue by wolfy1961** [llvm/llvm-project#92620]
The C++ standard (in section 17.6.3.2, paragraph 13 in the latest working draft) mandates about **operator delete()** that "if a function without a size parameter is defined, the program should also define the corresponding function with a size parameter. If a function with a size parameter is defined, the program shall also define the corresponding version without the size parameter".
Clang does not seem to issue a diagnostic if this is violated. Perhaps it should.
Example code:

```
int *p;
void operator delete(void* ptr) noexcept {
  (void) ptr;
  p = 0;
}

//  void operator delete( void* ptr, std::size_t size ) noexcept {
//      p = 0;
//  }

int s;
int main()
{
  p = new int;
  delete p;
}
```

**Comment by shafik** [comment]
No implementation diagnoses this: https://godbolt.org/z/bKoEo9d13

The first case uses *should* which is really *QoI* and the second one uses *shall* which makes me feel like this ends up being undefined behavior otherwise.

Maybe some library folks have strong opinions on what we should do here: @mordante @ldionne 

Also CC @AaronBallman

**Comment by wolfy1961** [comment]
The example came from a test suite. I'm not aware of anyone having a real world problem of this nature.

**Comment by ldionne** [comment]
Like @mordante , I don't see a problem with having such a diagnostic in principle. However, I think this diagnostic is basically impossible to implement without many false positives, since the `operator delete`s could be defined in different TUs to make the program valid, but the compiler can never know that.

**Comment by wolfy1961** [comment]
It does look like this can only be properly diagnosed at the linker level. I'll close this as won't fix. Please reopen if someone feels otherwise.

**Comment by ldionne** [comment]
Closing as wontfix per the above.
